### PR TITLE
Ubuntu 22.04 updates

### PR DESCRIPTION
**Pull request by faberga**
## Proposed changes

Ubuntu 22.04 updates to docker/Dockerfile.base

## Types of changes

What types of changes does your code introduce to the HE Toolkit project?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating
the PR. If you are unsure about any of them, do not hesitate to ask. We are
here to help! This is simply a reminder of what we are going to look for before
merging your code._

- [x] I have read the [CONTRIBUTING](https://github.com/intel/he-toolkit/tree/main#contributing) section
- [x] Current formatting and unit tests / base functionality passes locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

N/A

### CODE REVIEW

File: docker/Dockerfile.base [hunk 1]
```diff
@@ -21,18 +21,51 @@ ENV http_proxy=$http_proxy   \
...
-                   g++-10                         \
-                   gcc-10                         \
+                   xterm                          \
+                   cpuinfo                        \
+                   smbclient                      \
+                   silversearcher-ag              \
+                   zlib1g                         \
+                   zlib1g-dev                     \
+                   openjdk-11-jdk-headless        \
+                   sagemath                       \
```
Comment (hamishun) [inline docker/Dockerfile.base:68]: what are these for?

File: docker/Dockerfile.base [hunk 2]
```diff
@@ -43,24 +76,71 @@ RUN apt update && \
...
+      --slave /usr/bin/g++ g++ /usr/bin/g++-12                          \
+      --slave /usr/bin/gcov gcov /usr/bin/gcov-12                       \
+      --slave /usr/bin/gcc-ar gcc-ar /usr/bin/gcc-ar-12                 \
+      --slave /usr/bin/gcc-ranlib gcc-ranlib /usr/bin/gcc-ranlib-12     \
+      --slave /usr/bin/cpp cpp /usr/bin/cpp-12                          \
+      --slave /usr/bin/gcov-tool gcov-tool /usr/bin/gcov-tool-12        \
+      --slave /usr/bin/gcov-dump gcov-dump /usr/bin/gcov-dump-12        \
+      --slave /usr/bin/gcc-nm gcc-nm /usr/bin/gcc-nm-12                 \
+      --slave /usr/bin/lto-dump lto-dump /usr/bin/lto-dump-12
+# Update alternatives gor GCC-11
```
Comment (hamishun) [inline docker/Dockerfile.base:116]:
```suggestion
# Update alternatives for GCC-11
```

File: docker/Dockerfile.base (hunk 1 above)
Comment (faberga) [reply docker/Dockerfile.base:68]: these are for CNN application development

**Comment by faberga** [comment]
Closing this PR to fix a pletora of incompatibilities with newer versions of the pre-commit tools being used in Ubuntu 22.04.

**Pull request closed without merging**